pkg/apis/pipelinesascode/v1alpha1/types.go

```diff
@@ -70,11 +70,12 @@ type TaskInfos struct {
 
 // RepositorySpec is the spec of a repo
 type RepositorySpec struct {
-	ConcurrencyLimit *int         `json:"concurrency_limit,omitempty"`
-	URL              string       `json:"url"`
-	GitProvider      *GitProvider `json:"git_provider,omitempty"`
-	Incomings        *[]Incoming  `json:"incoming,omitempty"`
-	Params           *[]Params    `json:"params,omitempty"`
+	ConcurrencyLimit         *int         `json:"concurrency_limit,omitempty"`
+	URL                      string       `json:"url"`
+	GitProvider              *GitProvider `json:"git_provider,omitempty"`
+	Incomings                *[]Incoming  `json:"incoming,omitempty"`
+	Params                   *[]Params    `json:"params,omitempty"`
+	GithubAppTokenScopeRepos []string     `json:"github_app_token_scope_repos,omitempty"`
```

Comment:
I don't really like having this here since that's very specific to github app and rather not pollute this. 

If we start opening the door with multiple of fields it would get messy and ugly

what about in GitProvider as a sub struct

Reply:
Also `GitProvider` we don't use for GithubApp

Do you think adding another level under `GitProvider` specific to GithubApp ?

Reply:
great, i think you can remove that field tho now that you moved it to settings